esmvalcore/preprocessor/_ancillary_vars.py

```diff
@@ -0,0 +1,165 @@
+"""Preprocessor functions for ancillary variables and cell measures."""
+
+import logging
+import iris
+
+import dask.array as da
+
+from esmvalcore.preprocessor._io import load, concatenate_callback, concatenate
+from esmvalcore.cmor.fix import fix_metadata, fix_data
+from esmvalcore.cmor.check import cmor_check_metadata, cmor_check_data
+
+logger = logging.getLogger(__name__)
+
+
+def _load_fx(fx_info, check_level):
+    """Load and CMOR-check fx variables."""
+    fx_cubes = iris.cube.CubeList()
+
+    for fx_file in fx_info['filename']:
+        loaded_cube = load(fx_file, callback=concatenate_callback)
+        short_name = fx_info['short_name']
+        project = fx_info['project']
+        dataset = fx_info['dataset']
+        mip = fx_info['mip']
+        freq = fx_info['frequency']
+        loaded_cube = fix_metadata(loaded_cube, short_name=short_name,
+                                   project=project, dataset=dataset,
+                                   mip=mip, frequency=freq,
+                                   check_level=check_level)
+        fx_cubes.append(loaded_cube[0])
```

Comment:
loaded_cube is a CubeList right? why not check and extract the cube you need from the get-going and save a bit of resources?

Reply:
If the variables are time dependent they might be stored in different files. The fix metadata function works by file and then the cubes get concatenated to continue the checks. But I am not sure what do you mean here, sorry.

Reply:
yeah sorry, porr phrasing on my part - what I mean is `fx_file` can be loaded a cube with `load_cube()` since it'll never be a `CubeList` object (I believe) and then can be propagated though as a single cube. If this is not the case, and concatenation needs to be done on multiple cubes then I believe it's safer to concatenate using the `_io.concatenate()` function, so you don't risk concatenation fails due to issues with cube time overlaps or metadata

Reply:
>yeah sorry, poor phrasing on my part - what I mean is fx_file can be loaded a cube with load_cube() since it'll never be a CubeList object (I believe) and then can be propagated though as a single cube

Ah ok! @bouweandela preferred to use the `_io.load()` function though.

>I believe it's safer to concatenate using the _io.concatenate() function, so you don't risk concatenation fails due to issues with cube time overlaps or metadata.

Yes, the proper concatenation happens in the next line!

Reply:
indeed it does and I am fine with `_io.load()` too, it's more secure indeed, cheers Saskia! :+1: